sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/FileTypeSensorTest.java

```diff
@@ -0,0 +1,188 @@
+  }
+
+  @Test
+  public void whenProjectNameNotPresent_shouldNotAddProjectInfo() {
+    when(settingsMock.getString("sonar.projectName")).thenReturn(null);
+
+    sensor.execute(tester);
+
+    assertThat(logTester.logs()).isEmpty();
+    verifyZeroInteractions(projectTypeCollectorMock);
+  }
+
+  @Test
+  public void whenLanguageKeyIsPresent_logsCsOutputFile() {
+    when(settingsMock.getString("sonar.projectName")).thenReturn("FOO_PROJ");
+    when(settingsMock.getString("sonar.projectBaseDir")).thenReturn("BASE DIR");
+    when(settingsMock.getString("sonar.cs.analyzer.projectOutPath")).thenReturn("CS PATH");
+    when(settingsMock.getString("sonar.vbnet.analyzer.projectOutPath")).thenReturn("VB PATH");
+
+    when(pluginMetadataMock.languageKey()).thenReturn("cs");
+    sensor.execute(tester);
+    assertThat(logTester.logs(LoggerLevel.DEBUG)).containsExactly("Adding file type information (has MAIN 'false', has TEST 'false') for project 'FOO_PROJ' (base dir 'BASE DIR'). For debug info, see ProjectInfo.xml in 'CS PATH'.");
+
+    logTester.clear();
+    when(pluginMetadataMock.languageKey()).thenReturn("vbnet");
+    sensor.execute(tester);
+    assertThat(logTester.logs(LoggerLevel.DEBUG)).containsExactly("Adding file type information (has MAIN 'false', has TEST 'false') for project 'FOO_PROJ' (base dir 'BASE DIR'). For debug info, see ProjectInfo.xml in 'VB PATH'.");
+  }
+
+  @Test
+  public void whenProjectNamePresent_andHasNoFiles_shouldAddCorrectInfo() {
+    when(settingsMock.getString("sonar.projectName")).thenReturn("FOO_PROJ");
+
+    sensor.execute(tester);
+
+    assertThat(logTester.logs()).hasSize(1);
+    // we haven't mocked the base dir and project out settings
+    assertThat(logTester.logs(LoggerLevel.DEBUG)).containsExactly("Adding file type information (has MAIN 'false', has TEST 'false') for project 'FOO_PROJ' (base dir ''). For debug info, see ProjectInfo.xml in ''.");
+    verify(projectTypeCollectorMock, times(1)).addProjectInfo( false, false);
+  }
+
+  @Test
+  public void whenProjectNamePresent_andHasOnlyTestFiles_shouldAddCorrectInfo() {
+    when(settingsMock.getString("sonar.projectName")).thenReturn("FOO_PROJ");
+    addFileToFileSystem("foo.language", Type.TEST);
+
+    sensor.execute(tester);
+
+    assertThat(logTester.logs()).hasSize(1);
+    // we haven't mocked the base dir and project out settings
+    assertThat(logTester.logs(LoggerLevel.DEBUG)).containsExactly("Adding file type information (has MAIN 'false', has TEST 'true') for project 'FOO_PROJ' (base dir ''). For debug info, see ProjectInfo.xml in ''.");
+    verify(projectTypeCollectorMock, times(1)).addProjectInfo( false, true);
+  }
+
+  @Test
+  public void whenProjectNamePresent_andHasOnlyMainFiles_shouldAddCorrectInfo() {
+    when(settingsMock.getString("sonar.projectName")).thenReturn("FOO_PROJ");
+    addFileToFileSystem("foo.language", Type.MAIN);
+
+    sensor.execute(tester);
+
```

Comment:
Let's add log assert with `has MAIN 'true', has TEST 'false'` to show that it can happen and it should be there.

Reply:
This one is actually just to keep the test structure regular. The next one below is more interesting.